### PR TITLE
finished assignment

### PR DESCRIPTION
**Pull request by 0xf24**

### CODE REVIEW

File: Problem2/src/problem2/Problem2.java [hunk 1]
```diff
@@ -10,12 +10,39 @@
...
+     * sums up all of the digits, if they total to 3, 6, or 9, returns true,
+     * else false if the number is less than ten, calls self until base cases met
+     * @param d the number to check
+     * @return true if divisible, else false
      */
-    public static void main(String[] args) {
-        // TODO code application logic here
+    public static boolean isDivisibleByThree(int d) {
+        //check the base cases of 3, 6, and 9
+        if (d == 3 ||3 == 6 || 3 == 9) {
```
Comment (cgallinaro) [inline Problem2/src/problem2/Problem2.java:23]: You if statement should read "d == 3 || d == 6 || d == 9".  The tests you wrote fail because of this typo.

File: Problem2/test/problem2/Problem2Test.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+/**
+ *
+ * @author manxboy
+ */
+public class Problem2Test {
+    
+    @Test
+    public void testIsDivisibleByThree() {
```
Comment (cgallinaro) [inline Problem2/test/problem2/Problem2Test.java:19]: When testing recursive methods, test cases should cover base cases (3, 6, 9)

File: Problem3/test/problem3/PersonTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        assertTrue(apple.compareTo(ziglo) < 0);
+        
+        assertTrue(ziglo.compareTo(apple) > 0);
+        
+        assertTrue(dead.compareTo(ziglo) < 0);
+    }
+    
+    
+    @Test
+    public void testSort() {
```
Comment (cgallinaro) [inline Problem3/test/problem3/PersonTest.java:47]: Your testing should include edge cases like empty arrays and arrays of a single element.

File: Problem4/src/problem4/Problem4.java [hunk 4]
```diff
@@ -10,12 +10,35 @@
...
-    public static void main(String[] args) {
-        // TODO code application logic here
+    public static void sort(int[] data) {
+        
+        //loop over all entries backwards
+        for (int i = data.length - 1; i >= 1; i--) {
+            
+            //store the best value and best index
+            int best_index = -1;
+            int best_value = -1;
```
Comment (cgallinaro) [inline Problem4/src/problem4/Problem4.java:26]: If your array contains integers < -1, the initial value you are using will not work.  When testing your algorithm, I used the array [-2, -5, -4].  Your algorithm threw an exception because it tried to access index -1 in your array.

File: Problem4/test/problem4/Problem4Test.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import org.junit.Test;
+
+/**
+ *
+ * @author manxboy
+ */
+public class Problem4Test {
+    
+    @Test
+    public void testSort() {
```
Comment (cgallinaro) [inline Problem4/test/problem4/Problem4Test.java:19]: Your testing should include edge cases like empty arrays and arrays of a single element.

File: Problem5/src/problem5/Problem5.java [hunk 6]
```diff
@@ -5,17 +5,158 @@
...
+        
+        //set the end to -1
+        int end = -1;
+        
+        //loop until end != 1
+        for (int i = 1; end == -1; i++) {
+            //get the index of the next letter
+            int index = (first - 'a') + i;
+            
+            //if index excedes 26 return -1 "error"
```
Comment (cgallinaro) [inline Problem5/src/problem5/Problem5.java:103]: I don't think this comment is describing what the code is actually doing.

File: Problem5/test/problem5/Problem5Test.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author manxboy
+ */
+public class Problem5Test {
+    @Test
+    public void testSearch() {
```
Comment (cgallinaro) [inline Problem5/test/problem5/Problem5Test.java:17]: Should have additional tests that don't use the provided word list.  It would allow you cover edge cases like:
- empty array
- single word array
- word list where there are no letter starting with a particular character (i.e. no words that start with 'c')

**Comment by cgallinaro** [comment]
Problem 1: 5/5
Problem 2: 4/5
Problem 3: 4.5/5
Problem 4: 8/10
Problem 5: 9/10

Total: 30.5/35
